DQMOffline/Trigger/src/HLTTauDQMOfflineSource.cc

```diff
@@ -74,20 +77,45 @@ void HLTTauDQMOfflineSource::dqmBeginRun(const edm::Run& iRun, const edm::EventS
       }
       std::sort(foundPaths.begin(), foundPaths.end());
 
-      // Construct path plotters
-      std::vector<const HLTTauDQMPath *> pathObjects;
-      pathPlotters_.reserve(foundPaths.size());
-      pathObjects.reserve(foundPaths.size());
-      for(const std::string& pathName: foundPaths) {
-        pathPlotters_.emplace_back(pathName, HLTCP_, doRefAnalysis_, dqmBaseFolder_, hltProcessName_, nPtBins_, nEtaBins_, nPhiBins_, ptMax_, highPtMax_, l1MatchDr_, hltMatchDr_);
-        if(pathPlotters_.back().isValid()) {
-          pathObjects.push_back(pathPlotters_.back().getPathObject());
+
+      if(tagAndProbePaths.size() > 0) {
+        tagandprobePlotters_.reserve(tagAndProbePaths.size());
+        for(const edm::ParameterSet& tpset: tagAndProbePaths) {
+
+          const boost::regex tagpath(tpset.getUntrackedParameter<std::string>("tag"));
+          const boost::regex probepath(tpset.getUntrackedParameter<std::string>("probe"));
+          std::string xvariable = tpset.getUntrackedParameter<std::string>("xvariable");
+
+          std::string pathNameNum = "";
```

Comment:
would it be possible to make use of the `TriggerGenericEventFlag` [1] ?
(as it is done by DQMOffline/Trigger/plugins/METMonitor.cc)

thanks

[1]
CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h